lib/compress/zstd_compress.c

```diff
@@ -1492,11 +1520,12 @@ static size_t ZSTD_estimateCCtxSize_usingCCtxParams_internal(
         const ZSTD_paramSwitch_e useRowMatchFinder,
         const size_t buffInSize,
         const size_t buffOutSize,
-        const U64 pledgedSrcSize)
+        const U64 pledgedSrcSize,
+        int useExternalMatchFinder)
 {
     size_t const windowSize = (size_t) BOUNDED(1ULL, 1ULL << cParams->windowLog, pledgedSrcSize);
     size_t const blockSize = MIN(ZSTD_BLOCKSIZE_MAX, windowSize);
-    U32    const divider = (cParams->minMatch==3) ? 3 : 4;
+    U32    const divider = (cParams->minMatch==3 || useExternalMatchFinder) ? 3 : 4;
```

Comment:
minor : the logic employed to process `divider` seems re-used in another function later on.
It seems it would be better to factor it into a named function.